src/lasdi/latent_space.py

```diff
@@ -1,175 +1,482 @@
+        the network (the input to the first layer). If reshape_index is -1, then reshape_shape 
+        specifies how we reshape the network output before returning it (the output to the last 
+        layer). 
+
+        reshape_shape: These are lists of k integers specifying the final k dimensions of the shape
+        of the input to the first layer (if reshape_index == 0) or the output of the last layer 
+        (if reshape_index == -1). You must specify this argument if and only if you specify 
+        reshape_index. 
 
-        # Reshape input or output layer
+        threshold: You should only specify this argument if you are using the "Threshold" 
+        activation type. In this case, it specifies the threshold value for that activation (if x
+        is the input and x > threshold, then the function returns x. Otherwise, it returns the 
+        value).
+
+        value: You should only specify this argument if you are using the "Threshold" activation
+        type. In this case, "value" specifies the value that the function returns if the input is
+        below the threshold. 
+
+
+        -------------------------------------------------------------------------------------------
+        Returns
+        -------------------------------------------------------------------------------------------
+        
+        Nothing!
+        """
+
+        # Run checks.
+        # Make sure the reshape index is either 0 (input to 1st layer) or -1 (output of last 
+        # layer). Also make sure that that product of the dimensions in the reshape_shape match
+        # the input dimension for the 1st layer (if reshape_index == 0) or the output dimension of
+        # the last layer (if reshape_index == 1). 
+        # 
+        # Why do we need the later condition? Well, suppose that reshape_shape has a length of k. 
+        # If reshape_index == 0, then we squeeze the final k dimensions of the input and feed that 
+        # into the first layer. Thus, in this case, we need the last dimension of the squeezed 
+        # array to match the input domain of the first layer. On the other hand, reshape_index == -1 
+        # then we reshape the final dimension of the output to match the reshape_shape. Thus, in 
+        # both cases, we need the product of the components of reshape_shape to match a 
+        # corresponding element of layer_sizes.
         assert((reshape_index is None) or (reshape_index in [0, -1]))
         assert((reshape_shape is None) or (np.prod(reshape_shape) == layer_sizes[reshape_index]))
-        self.reshape_index = reshape_index
-        self.reshape_shape = reshape_shape
 
-        # Initalize activation function
-        self.act_type = act_type
-        self.use_multihead = False
+        super(MultiLayerPerceptron, self).__init__()
+
+        # Note that layer_sizes specifies the dimensionality of the domains and co-domains of each
```

Comment:
This comment is not parsed by API document. This can be parsed if the comment is written after the code with `""" ... """` comment as below:

```
self.n_layers : int = len(layer_sizes) - 1;
"""
Note that layer_sizes specifies the dimensionality of the domains and co-domains of each
layer. Specifically, the i'th element specifies the input dimension of the i'th layer,

...
"""
```